python-checks/src/main/java/org/sonar/python/checks/UndefinedNameAllPropertyCheck.java

```diff
@@ -0,0 +1,163 @@
+        checkAllProperty(ctx, assignmentStatement);
+      }
+    });
+  }
+
+  private static void checkAllProperty(SubscriptionContext ctx, AssignmentStatement assignmentStatement) {
+    Expression assignedValue = assignmentStatement.assignedValue();
+    if (!assignedValue.is(Tree.Kind.LIST_LITERAL) && !assignedValue.is(Tree.Kind.TUPLE)) {
+      return;
+    }
+    List<Tree> stringExpressions = getStringExpressions(assignedValue);
+    FileInput fileInput = (FileInput) TreeUtils.firstAncestorOfKind(assignedValue, Tree.Kind.FILE_INPUT);
+    if (fileInput == null) {
+      // Should never happen
+      return;
+    }
+    UnresolvedWildcardImportVisitor unresolvedWildcardImportVisitor = new UnresolvedWildcardImportVisitor();
+    fileInput.accept(unresolvedWildcardImportVisitor);
+    if (unresolvedWildcardImportVisitor.hasUnresolvedWildcardImport) {
+      return;
+    }
+    Map<String, Symbol> symbolsByName = fileInput.globalVariables().stream().collect(Collectors.toMap(Symbol::name, Function.identity()));
+    for (Tree stringExpression : stringExpressions) {
+      StringLiteral stringLiteral = retrieveStringLiteral(stringExpression);
+      if (isUnknownSymbol(ctx.pythonFile(), symbolsByName, stringLiteral)) {
+        PreciseIssue issue = ctx.addIssue(stringExpression, String.format("Change or remove this string; \"%s\" is not defined.", stringLiteral.trimmedQuotesValue()));
+        if (stringExpression != stringLiteral) {
+          issue.secondary(stringLiteral, null);
+        }
+      }
+    }
+  }
+
+  private static boolean isUnknownSymbol(PythonFile pythonFile, Map<String, Symbol> symbolsByName, StringLiteral stringLiteral) {
+    if (stringLiteral.stringElements().stream().anyMatch(StringElement::isInterpolated)) {
+      return false;
+    }
+    return !symbolsByName.containsKey(stringLiteral.trimmedQuotesValue())
+      && !BuiltinSymbols.all().contains(stringLiteral.trimmedQuotesValue())
+      && !isInitFileExportingModule(pythonFile, stringLiteral.trimmedQuotesValue());
+  }
+
+  /**
+   * As __init__.py files can declare modules which they don't import in their __all__ property, we need to exclude those potential FPs
+   */
+  private static boolean isInitFileExportingModule(PythonFile pythonFile, String name) {
+    return pythonFile.fileName().startsWith("__init__") && existsFileWithName(pythonFile.uri(), name);
+  }
+
+  private static List<Tree> getStringExpressions(Expression expression) {
+    if (expression.is(Tree.Kind.LIST_LITERAL)) {
+      return ((ListLiteral) expression).elements().expressions().stream()
+        .filter(UndefinedNameAllPropertyCheck::isString)
+        .collect(Collectors.toList());
+    }
+    return ((Tuple) expression).elements().stream()
+      .filter(UndefinedNameAllPropertyCheck::isString)
+      .collect(Collectors.toList());
+  }
+
+  private static class UnresolvedWildcardImportVisitor extends BaseTreeVisitor {
```

Comment:
_minor_ I think this class should be placed elsewhere, either to the top/bottom of the enclosing class or before/after the method that uses it.